<html><head><meta charset="utf-8"></meta></head><body><h1 class="article-title" style="text-align:center;">Scala throws关键字</h1><div style="width:100%;float:left;" class="article-content">   
 <p>Scala提供了<code>throws</code>关键字来声明异常。可以使用方法定义声明异常。 它向调用者函数提供了此方法可能引发此异常的信息。 它有助于调用函数处理并将该代码包含在<code>try-catch</code>块中，以避免程序异常终止。在scala中，可以使用<code>throws</code>关键字或<code>throws</code>注释来声明异常。</p> 
 <p><strong>Scala Throws示例</strong></p> 
 <pre><code class="lang-scala">class ExceptionExample4{  
    @throws(classOf[NumberFormatException])  
    def validate()={  
        "abc".toInt  
    }  
}  

object Demo{  
    def main(args:Array[String]){  
        var e = new ExceptionExample4()  
        try{  
            e.validate()  
        }catch{  
            case ex : NumberFormatException =&gt; println("Exception handeled here")  
        }  
        println("Rest of the code executing...")  
    }  
}
</code></pre> 
 <p>将上面代码保存到源文件：<em>Demo.scala</em>中，使用以下命令编译并执行代码 - </p>   
 <pre><code class="lang-shell">D:\software\scala-2.12.3\bin&gt;scalac Demo.scala
D:\software\scala-2.12.3\bin&gt;scala Demo.scal
Exception handeled here
Rest of the code executing...
</code></pre>
 <br>      
</div></body></html>